tests/benchmarks/object.cpp

```diff
@@ -209,30 +261,415 @@ TEST_CASE("Benchmark object", "[benchmark]") {
 
     SECTION("change notifications reporting") {
         auto table = r->read_group().get_table("class_person");
-        Results result(r, *table);
+        Results result(r, table);
         size_t num_calls = 0;
         size_t num_insertions = 0;
+        size_t num_deletions = 0;
+        size_t num_modifications = 0;
         auto token = result.add_notification_callback([&](CollectionChangeSet c, std::exception_ptr) {
             num_insertions += c.insertions.count();
+            num_deletions += c.deletions.count();
+            num_modifications += c.modifications_new.count();
             ++num_calls;
         });
 
         advance_and_notify(*r);
-        int64_t pk = 0;
+        ObjectSchema person_schema = *r->schema().find("person");
+        constexpr size_t num_objects = 1000;
```

Comment:
Only 1k objects probably isn't very interesting as even the old version is fine for small numbers of objects and it'd only become an issue as it scales.